### ISSUE TITLE
Not compatible with silverstripe/userforms >=5.7

### ISSUE THREAD

**Issue by jakxnz** [silverstripe/silverstripe-spamprotection#77]
# Versions

```
silverstripe/spamprotection 3.0.3
```

# Issue

An update in `SilverStripe\UserForms\Model\EditableFormField::onBeforeWrite()` has caused a compatibility issue with `SilverStripe\SpamProtection\EditableSpamProtectionField::onBeforeWrite()`, which can sometimes result in:

## Steps to reproduce

1. Create an `EditableSpamProtectionField` and attempt to `write` it without a Parent. i.e By creating via the code, or via an indirect means like a `populate` task

## Expected result

The object can be written

## Actual result

The object is not written, and an error is thrown:

```
ERROR [Emergency]: Uncaught Error: Call to a member function Fields() on null
IN GET dev/tasks/populate-task
Line 117 in /var/www/mysite/www/vendor/silverstripe/spamprotection/code/EditableSpamProtectionField.php
```

### PR
- [x] https://github.com/silverstripe/silverstripe-spamprotection/pull/79

**Comment by emteknetnz** [comment]
@jakxnz are you able to point to the exact code change in a pull-request that lead to this regression?

> which can sometimes result

What conditions are required for this to break?

**Comment by jakxnz** [comment]
@emteknetnz , I believe this is the change https://github.com/silverstripe/silverstripe-userforms/pull/1016

I've added steps to reproduce to this issue's description.

**Comment by maxime-rainville** [comment]
Fixed has been merged and I've tagged 3.1.1 and 3.2.1.
